### ISSUE TITLE
When should I update to the latest edition?

### ISSUE THREAD

**Issue by intgr** [rust-lang/edition-guide#268]
I feel like the guide should answer this question, but it doesn't: when does it make sense to update my packages to a new edition? What are the downsides and other considerations? In particular, does it restrict the range of Rust versions that can be used to build my package?

In fact, after a significant amount of Googling I still don't have clear answers to these questions.

**Comment by nikomatsakis** [comment]
I agree with @rylev -- the only reason I can see *not* to upgrade is MSRV. I would like to see people upgrade in general, but of course the main reason *to* upgrade would be to get access to the "nicer Rust" that it provides (sometimes there are new features that build directly on syntactic space reserved by the edition, of course).
